src/libs/actions/IOU.ts

```diff
@@ -2718,78 +2810,87 @@ function deleteMoneyRequest(transactionID, reportAction, isSingleTransactionView
                 },
             },
         },
-        {
-            onyxMethod: shouldDeleteIOUReport ? Onyx.METHOD.SET : Onyx.METHOD.MERGE,
-            key: `${ONYXKEYS.COLLECTION.REPORT}${iouReport.reportID}`,
-            value: iouReport,
-        },
+        shouldDeleteIOUReport
+            ? {
+                  onyxMethod: Onyx.METHOD.SET,
+                  key: `${ONYXKEYS.COLLECTION.REPORT}${iouReport?.reportID}`,
+                  value: iouReport,
+              }
+            : {
+                  onyxMethod: Onyx.METHOD.MERGE,
+                  key: `${ONYXKEYS.COLLECTION.REPORT}${iouReport?.reportID}`,
+                  value: iouReport ?? {},
```

Comment:
@aimane-chnaif It will be turned back just to `iouReport` here and in other similar places after [this PR](https://github.com/Expensify/react-native-onyx/pull/456) will be merged, so let's wait for that, sorry for the inconvenience